# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: f2f431ee78a3
# Full Hash: f2f431ee78a35a577f8d89a469ab17c20f71e3f0
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 03:56:08
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 2: Make `TextEditor` and `HTMLEditor` implement `EditorBase::CollapseSelectionToEndOfLastLeafNode` by themselves r=m_kato
#   
#   It does different thing for `TextEditor` and `HTMLEditor`, and used almost
#   internally.  Therefore, it should be implemented in the sub classes and
#   we should name them better.
# ==============================================================================

diff -r edac234f1eba -r f2f431ee78a3 editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Wed Sep 21 00:20:26 2022 +0000
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Wed Sep 21 00:38:23 2022 +0000
@@ -144,9 +144,11 @@
   //     removed by the web app and if they call `Selection::AddRange()`,
   //     it may cause multiple selection ranges.
   if (!SelectionRef().RangeCount()) {
-    nsresult rv = CollapseSelectionToEndOfLastLeafNode();
+    nsresult rv = CollapseSelectionToEndOfLastLeafNodeOfDocument();
     if (NS_FAILED(rv)) {
-      NS_WARNING("EditorBase::CollapseSelectionToEndOfLastLeafNode() failed");
+      NS_WARNING(
+          "HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument() "
+          "failed");
       return rv;
     }
   }